### PR TITLE
STITCH-1296 iOS: StitchCore/StitchCore-iOS Code Coverage Tooling

### PR DESCRIPTION
**Pull request by jsflax**

### CODE REVIEW

File: .evg.yml [hunk 1]
```diff
@@ -3,100 +3,196 @@ functions:
...
+      params:
+        shell: "bash"
+        script: |
+          set -e
+          echo "cloning stitch"
+          mkdir -p src/github.com/10gen
+          git clone git@github.com:10gen/stitch src/github.com/10gen/stitch
 
+          echo "downloading libmongo"
+          wget --quiet "https://s3.amazonaws.com/stitch-artifacts/stitch-mongo-libs/stitch_mongo_libs_linux_64_patch_1f02b06ec6fd48bad0b725b77bc194c84f839f53_5ab7216fc9ec44724e08f194_18_03_25_04_11_28/libmongo.so"
```
Comment (edaniels) [inline .evg.yml]: This is a linux shared object that isn't available. You want these two instead:
https://s3.amazonaws.com/stitch-artifacts/stitch-mongo-libs/stitch_mongo_libs_osx_patch_cbcbfd8ebefcca439ff2e4d99b022aedb0d61041_59e2b7a5c9ec4432c400181c_17_10_15_01_19_33/update_doc
https://s3.amazonaws.com/stitch-artifacts/stitch-mongo-libs/stitch_mongo_libs_osx_patch_cbcbfd8ebefcca439ff2e4d99b022aedb0d61041_59e2b7ab2a60ed5647001827_17_10_15_01_19_39/assisted_agg

both need to be chmod +x

File: .evg.yml [hunk 2]
```diff
@@ -3,100 +3,196 @@ functions:
...
+          chmod +x libmongo.so
+
+          echo "building transpiler"
+          cd src/github.com/10gen/stitch/etc/transpiler
+          yarn install && yarn run build -t ${transpiler_target}
```
Comment (edaniels) [inline .evg.yml]: This may not be available on the machine. Might need to file a BUILD

File: .evg.yml [hunk 3]
```diff
@@ -3,100 +3,196 @@ functions:
...
-
-  - name: github_pull_request_end
-    depends_on:
-      - name: github_pull_request_start
-      - name: run_tests
-        status: "*"
-        variant: linux-64
+            set -e
+            cd stitch-ios-sdk/StitchCore
+            swiftlint
```
Comment (edaniels) [inline .evg.yml]: [q] Can we run a script that lints everything, not just core? That's what java is doing

File: .evg.yml [hunk 4]
```diff
@@ -3,100 +3,196 @@ functions:
...
-              if [ $? -ne 0 ]; then
-                echo "failure" > commit_state
-                exit 0
-              fi
-            done
-
-            echo "success" > commit_state
-      - func: "set_github_build_status"
+            set -e
+            cd stitch-ios-sdk/StitchCore
```
Comment (edaniels) [inline .evg.yml]: [q] Can we run a script that tests everything, not just core? That's what java is doing

File: .evg.yml [hunk 5]
```diff
@@ -3,100 +3,196 @@ functions:
...
 
-  - name: run_tests
+            echo "running swift core tests"
+            
+            swift package generate-xcodeproj
+            xcodebuild test -scheme SwiftCode-Package -enableCodeCoverage YES
+            xcov --legacy_support xccoverage --source_directory Sources/ --exclude_targets ExtendedJSON.framework,JWT.framework,StitchCore_AdminClient.framework,Swifter.framework
```
Comment (edaniels) [inline .evg.yml]: [nit] We should be including extjson and admin client

File: .evg.yml [hunk 6]
```diff
@@ -3,100 +3,196 @@ functions:
...
-            echo $RESULT | grep '!!ALL TESTS PASSED!!'
+            echo "running iOS tests"
+            xcodebuild test -workspace ../Stitch.xcworkspace/ -scheme StitchCore-iOS -configuration Debug -enableCodeCoverage YES -destination "platform=iOS Simulator,name=iPhone 8,OS=11.2"
+            xcov --legacy_support xccoverage --exclude_targets ExtendedJSON.framework,JWT.framework,StitchCore_AdminClient.framework,Swifter.framework
+      - func: "publish_codecov"
+post:
+  - command: attach.xunit_results
+    params:
+      files:
+        - "./stitch-ios-sdk/StitchCore/xcov_report/**/*"
```
Comment (edaniels) [inline .evg.yml]: Just do one `"./stitch-ios-sdk/**/xcov_report/**/*"`

File: .evg.yml [hunk 7]
```diff
@@ -3,100 +3,196 @@ functions:
...
+        - "./stitch-ios-sdk/StitchCore-iOS/xcov_report/**/*"
```
Comment (edaniels) [inline .evg.yml]: [q] Are all of the files guaranteed to have unique names? If not, then you will keep uploading the same file.

File: .evg.yml [hunk 8]
```diff
@@ -3,100 +3,196 @@ functions:
...
+  - command: s3.put
+    params:
+      aws_key: ${test_artifacts_aws_key}
+      aws_secret: ${test_artifacts_aws_secret}
+      remote_file: ${build_id}/report
```
Comment (edaniels) [inline .evg.yml]: [major] Missing `local_files_include_filter`. See java evergreen file. You need to make sure each file has a unique base name as well. It gets concatenated to the remote_file base name.

File: .evg.yml [hunk 9]
```diff
@@ -3,100 +3,196 @@ functions:
...
+      content_type: text/html
+      permissions: public-read
 
 buildvariants:
 - name: linux-64
   display_name: Linux-64
   run_on:
     - baas-linux
+  expansions:
+    mongodb_url: http://fastdl.mongodb.org/linux/mongodb-linux-x86_64-3.4.4.tgz
```
Comment (edaniels) [inline .evg.yml]: [major] This is not linux.

File: .evg.yml [hunk 10]
```diff
@@ -3,100 +3,196 @@ functions:
...
+    transpiler_target: node8-linux
+  tasks:
+    - name: lint
+    - name: run_core_tests
+- name: macos-1012
+  display_name: MacOS 1012
+  run_on:
+    - macos-1012
+  expansions:
+    mongodb_url: http://fastdl.mongodb.org/linux/mongodb-linux-x86_64-3.4.4.tgz
```
Comment (edaniels) [inline .evg.yml]: [major] This is not linux.

File: .evg.yml [hunk 11]
```diff
@@ -3,100 +3,196 @@ functions:
...
+    params:
+      aws_key: ${test_artifacts_aws_key}
+      aws_secret: ${test_artifacts_aws_secret}
+      remote_file: ${build_id}/report
+      bucket: stitch-test-artifacts
+      content_type: text/html
+      permissions: public-read
 
 buildvariants:
 - name: linux-64
```
Comment (edaniels) [inline .evg.yml]: This variant should not exist

File: .evg.yml [hunk 12]
```diff
@@ -3,100 +3,196 @@ functions:
...
+  run_on:
+    - macos-1012
+  expansions:
+    mongodb_url: http://fastdl.mongodb.org/linux/mongodb-linux-x86_64-3.4.4.tgz
+    transpiler_target: node8-linux
   tasks:
-    - name: github_pull_request_start
-    - name: run_tests
-    - name: github_pull_request_end
+    - name: run_ios_tests
```
Comment (edaniels) [inline .evg.yml]: This should run all of the tests.

File: .evg.yml [hunk 13]
```diff
@@ -3,100 +3,196 @@ functions:
...
+  tasks:
+    - name: lint
+    - name: run_core_tests
+- name: macos-1012
+  display_name: MacOS 1012
+  run_on:
+    - macos-1012
+  expansions:
+    mongodb_url: http://fastdl.mongodb.org/linux/mongodb-linux-x86_64-3.4.4.tgz
+    transpiler_target: node8-linux
```
Comment (edaniels) [inline .evg.yml]: You need to change this target for mac

**Comment by jsflax** [comment]
Consider this scrapped for now.

**Pull request closed without merging**